src/test/java/com/devonfw/sample/archunit/ComponentRuleTest.java

```diff
@@ -0,0 +1,263 @@
+package com.devonfw.sample.archunit;
+
+import com.tngtech.archunit.lang.ArchCondition;
+import com.tngtech.archunit.lang.ArchRule;
+import com.tngtech.archunit.lang.ConditionEvents;
+import com.tngtech.archunit.lang.SimpleConditionEvent;
+
+import static com.tngtech.archunit.library.dependencies.SlicesRuleDefinition.slices;
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.noClasses;
+
+import com.tngtech.archunit.core.domain.Dependency;
+import com.tngtech.archunit.core.domain.JavaClass;
+import com.tngtech.archunit.core.importer.ImportOption;
+import com.tngtech.archunit.junit.AnalyzeClasses;
+import com.tngtech.archunit.junit.ArchTest;
+
+import static com.tngtech.archunit.base.DescribedPredicate.alwaysTrue;
+import static com.tngtech.archunit.core.domain.properties.HasName.Predicates.nameMatching;
+
+@AnalyzeClasses(packages = "com.devonfw.sample.archunit", importOptions = ImportOption.DoNotIncludeTests.class)
+public class ComponentRuleTest {
+
+    /**
+     * verifying that the service layer of one component does not depend on the
+     * service layer of
+     * another component.
+     */
+    @ArchTest
+    public static final ArchRule C3_no_service_layer_depends_on_service_layer_of_another_component = slices()
+            .matching("..archunit.(*).service..")
+            .namingSlices("$1 service").should().notDependOnEachOther()
+            .ignoreDependency(alwaysTrue(), nameMatching(".*general.*"))
+            .as("Code from service layer shall not depend on service layer of a different component");
+
+    public static final ArchCondition<JavaClass> haveComponentServiceLayerDependingOnDiffComponentsLogicLayer = new ArchCondition<JavaClass>(
+            "have dependencies, towards another components logic layer (Rule-C4).") {
+        @Override
+        public void check(JavaClass sourceClass, ConditionEvents events) {
+
+            String sourceClassPackageName = sourceClass.getPackageName();
+            PackageStructure sourcePkg = PackageStructure.of(sourceClassPackageName);
+            String sourceClassLayer = sourcePkg.getLayer();
+            String sourceClassComponent = sourcePkg.getComponent();
+            // Check all project components' service layers except the default component for
+            // noncompliant dependencies towards
+            // other components' logic layers.
+            if (sourcePkg.isLayerService() && !sourceClassComponent.equals("")
```

Comment:
FYI: I do not see any case how `isLayerService()` could be `true` while  `sourceClassComponent.equals("")` as Java does not allow empty package segments and OS also does not allow folders with empty names.

Reply:
Deleted the `sourceClassComponent.equals("")` conditions throughout all rules.